distributed/recreate_exceptions.py

```diff
@@ -124,6 +131,13 @@ def recreate_error_locally(self, future):
         Nothing; the function runs and should raise an exception, allowing
         the debugger to run.
         """
-        func, args, kwargs = sync(
-                self.client.loop, self._recreate_error_locally, future)
-        func(*args, **kwargs)
+        out = sync(self.client.loop, self._recreate_error_locally, future)
+        if len(out) == 3 and isinstance(out[2], dict):
+            func, args, kwargsfunc(*args, **kwargs)
+        else:
+            exec_tuple(out)
+
+
+def exec_tuple(t):
+    newt = (exec_tuple(v) if isinstance(v, tuple) else v for v in t[1:])
+    return t[0](*newt)
```

Comment:
This feels like it might not handle all corner cases.  I recommend using `dask.get` here instead to ensure that we cover the full spec.  The relevant internal function is probably `dask/async.py::_execute_task`, though relying on public API (dask.get) is probably nicer.